packages/labs/signals/README.md

```diff
@@ -0,0 +1,208 @@
+
+const count = signal(0);
+
+@customElement('signal-example')
+export class SignalExample extends SignalWatcher(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${count.get()}</p>
+      <button @click=${this._onClick}>Increment<button></button></button>
+    `;
+  }
+
+  private _onClick() {
+    count.set(count.get() + 1);
+  }
+}
+```
+
+Elements should not _write_ to signals in these lifecycle methods or they might
+cause an infinite loop.
+
+### watch() directive
+
+The `watch()` directive accepts a single Signal and renders its value,
+subscribing to updates and updating the DOM when the signal changes.
+
+The `watch()` directive allows for very targeted updates of the DOM, which can
+be good for performance (but as always, measure!).
+
+Updates from `watch()` directives are batches and commited during the next
+reactive update. If there are only updates from `watch()` directives, then those
+updates are commited without a full render - preserving both DOM coherence and
+targetted updates. If there is another reactive update pending, then the whole
+template is re-rendered, along with the latest signal values.
+
+`watch()` must be used in conjunction with `SignalWatcher`.
+
+```ts
+import {LitElement, html} from 'lit';
+import {customElement, property} from 'lit';
+import {SignalWatcher, watch, signal} from '@lit-labs/preact-signals';
+
+const count = signal(0);
+
+@customElement('signal-example')
+export class SignalExample extends SignalWatcher(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${watch(count)}</p>
+      <button @click=${this._onClick}>Increment<button></button></button>
+    `;
+  }
+
+  private _onClick() {
+    count.set(count.get() + 1);
+  }
+}
+```
+
+You can mix and match targetted updates with `watch()` directive and auto-tracking with `SignalWatcher`. When you pass a signal directly to `watch()` it is not accessed in a callback
+watched by `SignalWatcher`, so an update to that signal will only cause a targeted
+DOM update and not an full template render.
+
+### html tag and withWatch()
+
+This package also exports an `html` template tag that can be used in place of
+Lit's default `html` tag and automatically wraps any signals in `watch()`.
+
+```ts
+import {LitElement} from 'lit';
+import {customElement, property} from 'lit';
+import {SignalWatcher, html, signal} from '@lit-labs/preact-signals';
+
+const count = signal(0);
+
+@customElement('signal-example')
+export class SignalExample extends SignalWatcher(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${count}</p>
+      <button @click=${this._onClick}>Increment<button></button></button>
+    `;
+  }
+
+  private _onClick() {
+    count.set(count.get() + 1);
+  }
+}
```

Comment:
the example doesnt include the `withWatch` function?

Reply:
Yeah, there's no `withWatch()` example here because the `html` tag exported by this package has already applied it. `withWatch()` is kind of advanced.

Reply:
Added `withWatch()` example.